{{cookiecutter.project_name}}/.github/workflows/pypi-publish.yaml

```diff
@@ -18,18 +18,17 @@ jobs:
         python-version: 3.9
 
     - name: Install Poetry
-      uses: snok/install-poetry@v1.1.6
-      with:
-        virtualenvs-create: true
-        virtualenvs-in-project: true
+      run: pip install poetry poetry-dynamic-versioning
```

Comment:
This installs poetry into the same env as the project env which can become a problem if poetry´s dependencies contradict the ones in your project. In this pipeline this is no problem because you don't need to install to build.

However, in the other pipelines where you do install

- https://github.com/linkml/linkml-project-cookiecutter/blob/main/%7B%7Bcookiecutter.project_name%7D%7D/.github/workflows/main.yaml
- https://github.com/linkml/linkml-project-cookiecutter/blob/main/%7B%7Bcookiecutter.project_name%7D%7D/.github/workflows/deploy-docs.yaml

I would suggest to use [pipx](https://pypa.github.io/pipx/) to achieve isolation between poetry and the project. For pipx (which is available by default) the poetry-install section would look like:
```yaml
    - name: Install Poetry
      run: |
        pipx install poetry
        pipx inject poetry poetry-dynamic-versioning
```

Reply:
I was under the impression that different workflows ('main.yaml' , `deploy-docs.yaml`) are contained in independent environments. Isn't that true?

Reply:
No problem. I would also pip install here. I just wanted to show the alternative.

Yes, the other workflows are isolated from this one. That is why I thought using pipx here has no advantage. It would only make sense in the other two workflows. But everyone weights the arguments pro/contra pipx that is better isolation vs. using yet another tool, differently.